File: 2206-divide-array-into-equal-pairs/2206-divide-array-into-equal-pairs.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var divideArray = function(nums) {
    const numCount = new Map();
    
    for(let i = 0; i < nums.length; i++) {
        if(numCount.has(nums[i])) {
            numCount.delete(nums[i]);
        } else {
            numCount.set(nums[i], true);
        }
    }
    return numCount.size === 0;
};

File: 2206-divide-array-into-equal-pairs/2206-divide-array-into-equal-pairs.py
class Solution:
    def divideArray(self, nums: List[int]) -> bool:
        count = {}
        pairs = 0
        for num in nums:
            count[num] = count.get(num, 0) + 1
        
        for value in count.values():
            if value % 2 != 0:
                return False
            
        return True